web/client/plugins/DeleteMap.jsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2022, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+*/
+
+import React from 'react';
+import PropTypes from "prop-types";
+import { Glyphicon } from 'react-bootstrap';
+import { connect } from 'react-redux';
+import { createSelector } from 'reselect';
+
+import ConfirmDialog from '../components/misc/ConfirmDialog';
+import { createPlugin } from '../utils/PluginsUtils';
+import { deleteMap } from '../actions/maps';
+import { toggleControl } from '../actions/controls';
+import { mapIdSelector } from '../selectors/mapInitialConfig';
+import Message from '../components/I18N/Message';
+
+class DeleteConfirmDialog extends React.Component {
+
+    static contextTypes = {
+        router: PropTypes.object
+    };
+
+    static defaultProps = {
+        show: false,
+        mapId: '',
+        onConfirmDelete: () => {},
+        onClose: () => {}
+    };
+
+    render() {
+        return (
+            <ConfirmDialog
+                show={this.props.show}
+                onClose={this.props.onClose}
+                title={<Message msgId="map.mapDelete" />}
+                onConfirm={() => {
+                    this.context.router.history.push("/");
+                    this.props.onConfirmDelete(this.props.mapId);
+                }}
+                fitContent
+            >
+                <div className="ms-detail-body">
+                    <Message msgId="resources.deleteConfirmMessage" />
+                </div>
+            </ConfirmDialog>
+        );
+    }
+}
+
+DeleteConfirmDialog.propTypes = {
+    show: PropTypes.bool,
+    mapId: PropTypes.string,
+    onConfirmDelete: PropTypes.func,
+    onClose: PropTypes.func
+};
+
```

Comment:
let's stick to one way of doing things this way of declaring PropTypes is mostly used for functional components
if you use classes let's move this inside the class component

Reply:
Reolved